api/v1alpha1/virtualmachine_conversion_test.go

```diff
@@ -541,6 +541,285 @@ func TestVirtualMachineConversion(t *testing.T) {
 			g.Expect(c.Reason).To(Equal("Reason"))
 			g.Expect(c.Message).To(Equal("Message"))
 		})
+
+		t.Run("Converts PrereqReady Condition", func(t *testing.T) {
+			t.Run("PrereqReady is True with no prior v1a2 conditions", func(t *testing.T) {
+				g := NewWithT(t)
+
+				now := metav1.Now()
+				spoke := v1alpha1.VirtualMachine{
+					Spec: v1alpha1.VirtualMachineSpec{
+						VmMetadata: &v1alpha1.VirtualMachineMetadata{
+							Transport:  v1alpha1.VirtualMachineMetadataCloudInitTransport,
+							SecretName: "my-secret",
+						},
+						ResourcePolicyName: "my-policy",
+					},
+					Status: v1alpha1.VirtualMachineStatus{
+						Conditions: []v1alpha1.Condition{
+							{
+								Type:               v1alpha1.VirtualMachinePrereqReadyCondition,
+								Status:             corev1.ConditionTrue,
+								LastTransitionTime: now,
+							},
+						},
+					},
+				}
+
+				hub := &nextver.VirtualMachine{}
+				g.Expect(spoke.ConvertTo(hub)).To(Succeed())
+				g.Expect(hub.Status.Conditions).To(HaveLen(4))
```

Comment:
nit: Add an assertion that the PreReqReady condition does not get set anymore. That's one of the things we are addressing with this patch.